### PR TITLE
Update workflow setup script for free-forecast ensemble 

### LINKED ISSUE #828: Update workflow setup script for free-forecast ensemble
**Description**
As we move towards the merger of GEFS into the GFS system, we need to be able to generate workflows with an arbitrary number of free-forecast ensemble members in addition to the number of EnKF members.

**Requirements**
Need to be able to read in a `config.base` file from the experiment directory and use the number of free-forecast ensemble members to generate ensemble metatasks.  Note this is just for generic capability, with specific ensemble task generation added as each task is expanded to support ensembles (see other issues in the [associated Epic](https://github.com/noaa-emc/global-workflow/issues/613)).

**Acceptance Criteria (Definition of Done)**
Functions/tools to properly generate ensemble metatasks given other task dependencies.

**Dependencies**
Consolidation of workflow setup scripts (#???)

### PR DESCRIPTION
**Pull request by XianwuXue-NOAA**
**Description**

As we move towards the merger of GEFS into the GFS system, this PR is to add new feature which to generate workflows with an arbitrary number of free-forecast ensemble members in addition to the number of GEFS members.

**Please include a summary of the change and which issue is fixed.**

 Improve ush/rocoto/setup_workflow_fcstonly.py to add some metatasks (getic, init and fcst) for GEFS to xml file.

**Type of change**

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

**How Has This Been Tested?**

It has been tested on Hera.

  
**Checklist**

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] Any dependent changes have been merged and published

Dep: #849

### CODE REVIEW

File: ush/rocoto/setup_workflow_fcstonly.py [hunk 1]
```diff
@@ -207,6 +219,13 @@ def get_postgroups(post, cdump='gdas'):
         fhout_hf = post['FHOUT_HF_GFS']
         fhrs_hf = list(range(fhmin, fhmax_hf + fhout_hf, fhout_hf))
         fhrs = fhrs_hf + list(range(fhrs_hf[-1] + fhout, fhmax + fhout, fhout))
+    elif cdump in ['gefs']:
```
Comment (WalterKolczynski-NOAA) [inline ush/rocoto/setup_workflow_fcstonly.py]: This looks to be duplicate of 'gfs'. Maybe just add 'gefs' to that list unless/until we create a separate set of variables?

File: ush/rocoto/setup_workflow_fcstonly.py [hunk 2]
```diff
@@ -299,40 +320,68 @@ def get_workflow(dict_configs, cdump='gdas'):
     else:
         if hpssarch in ['YES']:
             deps = []
-            data = '&ROTDIR;/&CDUMP;.@Y@m@d/@H/atmos/INPUT/sfc_data.tile6.nc'
+            if cdump == "gefs":
```
Comment (WalterKolczynski-NOAA) [inline ush/rocoto/setup_workflow_fcstonly.py]: To match the normal python idiom (and the rest of the code), these should all be `if cdump in ['gefs'].

File: ush/rocoto/setup_workflow_fcstonly.py [hunk 3]
```diff
@@ -458,8 +521,11 @@ def get_workflow(dict_configs, cdump='gdas'):
     vardict = {varname2: varval2, varname3: varval3}
     task = wfu.create_wf_task('post', cdump=cdump, envar=postenvars, dependency=dependencies,
                               metatask='post', varname=varname1, varval=varval1, vardict=vardict)
-    tasks.append(task)
-    tasks.append('\n')
+    if cdump == "gefs":
```
Comment (WalterKolczynski-NOAA) [inline ush/rocoto/setup_workflow_fcstonly.py]: Add a comment to these noting they are temporary bypasses for GEFS until we want to add these tasks to the GEFS task mesh.

**Review by WalterKolczynski-NOAA — CHANGES_REQUESTED** [review]

**Comment by WalterKolczynski-NOAA** [comment]
Like the other PR, this one needs to have its title and description filled out properly.

File: ush/rocoto/setup_workflow_fcstonly.py (hunk 1 above)
Comment (XianwuXue-NOAA) [reply ush/rocoto/setup_workflow_fcstonly.py]: Currently, I am not sure whether GEFS will need other variables, and gdas also has a block of statements, so I 'd like to keep this block and improve them in the future.

**Comment by XianwuXue-NOAA** [comment]
@WalterKolczynski-NOAA I just push two commits to make the changes based on your comments. For the rest comment, I'd like to keep the block for the future's improvement.

**Review by aerorahul — COMMENTED** [review]
I would like to review this in a little more detail.
Please hold from merging.

**Comment by aerorahul** [comment]
@XianwuXue-NOAA 
Please provide a description of what the changes are and what they intend to accomplish.

**Comment by XianwuXue-NOAA** [comment]
@aerorahul I changed ush/rocoto/setup_workflow_fcstonly.py to add some metatasks (getic, init and fcst) for GEFS to xml file.

**Comment by WalterKolczynski-NOAA** [comment]
> @aerorahul I changed ush/rocoto/setup_workflow_fcstonly.py to add some metatasks (getic, init and fcst) for GEFS to xml file.

You need a more complete description, and it needs to be in the first post, just like a regular commit message. It should describe what and why. And a relevant, less cryptic, title.

**Comment by XianwuXue-NOAA** [comment]
@WalterKolczynski-NOAA Thanks a lot. I will modify the first post.

**Pull request closed without merging**